src/new-client/src/App/Footer/StatusButton/StatusButton.tsx

```diff
@@ -81,28 +85,81 @@ const selectAll = (event: SyntheticEvent) => {
   input.select()
 }
 
+const getInstanceNumber = (
+  record: Record<string, ServiceInstanceStatus>,
+  type: string,
+) => {
+  return Object.values(record).filter(
+    (x) => x.isConnected && x.serviceType === type,
+  ).length
+}
+
+const timeoutThreshold = 3000
+const requestTimeoutLogger$ = of(
+  ["blotter", "reference", "execution", "pricing", "analytics"].map(
+    (serviceType) =>
+      ({
+        serviceType,
+        connectionStatus: ServiceConnectionStatus.DISCONNECTED,
+        connectedInstanceCount: 0,
+      } as ServiceStatus),
+  ),
+)
+
+export const [useServiceStatus, serviceStatus$] = bind(
+  status$.pipe(
+    map((record) => {
+      const nameList: Set<string> = new Set()
+      const result: ServiceStatus[] = []
+      const recordValues = Object.values(record)
+      recordValues.forEach((next) => {
+        const newStatus = {
+          serviceType: next.serviceType,
+          connectedInstanceCount: getInstanceNumber(record, next.serviceType),
+          connectionStatus: getInstanceNumber(record, next.serviceType)
+            ? ServiceConnectionStatus.CONNECTED
+            : ServiceConnectionStatus.DISCONNECTED,
+        }
+        if (nameList.size !== nameList.add(newStatus.serviceType).size) {
+          result.push(newStatus)
+        }
+      })
+      return result
+    }),
```

Comment:
I'm pretty sure that the only transformation that's needed here is `map(Object.values)`